### ISSUE TITLE
Problems submitting form data to new window 

### ISSUE THREAD

**Issue by toolyq** [MicrosoftEdge/WebView2Feedback#3203]
**Description**

Hi team,
The issue happend again.
Problems submitting form data to new window #1116
https://github.com/MicrosoftEdge/WebView2Feedback/issues/1116

**Version**
SDK: 1.0.1210.39
Runtime: 109.0.1518.78 
Framework: Win32
OS: Win10

**Repro Steps**
```
CHECK_FAILURE(m_webView->add_NewWindowRequested(
    Callback<ICoreWebView2NewWindowRequestedEventHandler>(
        [this](ICoreWebView2* sender, ICoreWebView2NewWindowRequestedEventArgs* args) {
            wil::com_ptr<ICoreWebView2Deferral> deferral;
            CHECK_FAILURE(args->GetDeferral(&deferral));
            AppWindow* newAppWindow;
            wil::com_ptr<ICoreWebView2WindowFeatures> windowFeatures;
            CHECK_FAILURE(args->get_WindowFeatures(&windowFeatures));

            RECT windowRect = {0};
            UINT32 left = 0;
            UINT32 top = 0;
            UINT32 height = 0;
            UINT32 width = 0;
            BOOL shouldHaveToolbar = true;

            BOOL hasPosition = FALSE;
            BOOL hasSize = FALSE;
            CHECK_FAILURE(windowFeatures->get_HasPosition(&hasPosition));
            CHECK_FAILURE(windowFeatures->get_HasSize(&hasSize));

            bool useDefaultWindow = true;

            if (!!hasPosition && !!hasSize)
            {
                CHECK_FAILURE(windowFeatures->get_Left(&left));
                CHECK_FAILURE(windowFeatures->get_Top(&top));
                CHECK_FAILURE(windowFeatures->get_Height(&height));
                CHECK_FAILURE(windowFeatures->get_Width(&width));
                useDefaultWindow = false;
            }
            CHECK_FAILURE(windowFeatures->get_ShouldDisplayToolbar(&shouldHaveToolbar));

            windowRect.left = left;
            windowRect.right = left + (width < s_minNewWindowSize ? s_minNewWindowSize : width);
            windowRect.top = top;
            windowRect.bottom = top + (height < s_minNewWindowSize ? s_minNewWindowSize : height);

            // passing "none" as uri as its a noinitialnavigation
            if (!useDefaultWindow)
            {
                newAppWindow = new AppWindow(
                    m_creationModeId, L"none", m_userDataFolder, false, nullptr, true, windowRect,
                    !!shouldHaveToolbar);
            }
            else
            {
              newAppWindow = new AppWindow(m_creationModeId, L"none");
            }
            newAppWindow->m_isPopupWindow = true;
            newAppWindow->m_onWebViewFirstInitialized = [args, deferral, newAppWindow]() {
                CHECK_FAILURE(args->put_NewWindow(newAppWindow->m_webView.get()));

                // listen to WebResourceRequested event for new webview once we set new webview as window.
                newAppWindow->m_webView.get()->AddWebResourceRequestedFilter(
                    L"*", COREWEBVIEW2_WEB_RESOURCE_CONTEXT_ALL);
                CHECK_FAILURE(newAppWindow->m_webView.get()->add_WebResourceRequested(
                    Callback<ICoreWebView2WebResourceRequestedEventHandler>(
                        [](
                            ICoreWebView2* sender,
                            ICoreWebView2WebResourceRequestedEventArgs* args) {
                                COREWEBVIEW2_WEB_RESOURCE_CONTEXT resourceContext;
                                CHECK_FAILURE(args->get_ResourceContext(&resourceContext));

                                wil::com_ptr<ICoreWebView2WebResourceRequest> request;
                                CHECK_FAILURE(args->get_Request(&request));

                                wil::unique_cotaskmem_string uri;
                                CHECK_FAILURE(request->get_Uri(&uri));

                                wil::unique_cotaskmem_string method;
                                CHECK_FAILURE(request->get_Method(&method)); // ****** The method is always GET **********

                                wil::com_ptr<IStream> content;
                                CHECK_FAILURE(request->get_Content(&content));

                                return S_OK;
                        })
                    .Get(), nullptr));

                CHECK_FAILURE(args->put_Handled(TRUE));
                CHECK_FAILURE(deferral->Complete());
            };

            return S_OK;
        })
        .Get(),
    nullptr));
```

**Screenshots**
![image](https://user-images.githubusercontent.com/12145763/217704087-bf92602a-60c2-47df-b904-0594febd4385.png)

**Additional context**
It workked at RUNTIME 95.0.1020.30 and 102.0.1245.53.

**Comment by toolyq** [comment]
![image](https://user-images.githubusercontent.com/12145763/217705379-52d5b0ec-9f61-4868-948e-a21348793aa4.png)

**Comment by novac42** [comment]
Thanks for the update. Tracking under #1116

**Comment by monica-ch** [comment]
@toolyq Thanks for the bug report. However, I was able to get the WRR event back and get_Method was `POST` using WebView2 runtime version you specified. What do you notice when you try in Edge browser?

Below code snippet is used to submit to a new window using POST.

```<form method="post" action="https://httpbin.org/post" target="_blank">
    <label>First name:</label>
    <input type="text" name="fname" value="test"><br>
    <label>Last name:</label>
    <input type="text" name="lname" value="test"><br>
    <input type="submit" value="Submit">
</form>
```

**Comment by toolyq** [comment]
@monica-ch ,it worked as expected in edge browser, but not in my code.

**Comment by toolyq** [comment]
Only if I downgrade to 102.0.1245.53, it worked.
![image](https://user-images.githubusercontent.com/12145763/217974353-59bfe8ca-6c80-4f86-a634-7258e4931026.png)

**Comment by toolyq** [comment]
Reproduce code:
``
HRESULT CMFC_webview2Dlg::OnCreateCoreWebView2ControllerCompleted(HRESULT result, ICoreWebView2Controller* controller)
{
	if (controller != nullptr) {
		m_controller = controller;
		m_controller->get_CoreWebView2(&m_webView);
	}
	else
	{
		return S_FALSE;
	}
	RECT bounds;
	GetClientRect(&bounds);
	m_controller->put_Bounds(bounds);

	EventRegistrationToken newWindowRequestedtoken;
	m_webView->add_NewWindowRequested(Microsoft::WRL::Callback<ICoreWebView2NewWindowRequestedEventHandler>(//wrap windown.open()
		[=](ICoreWebView2* webview, ICoreWebView2NewWindowRequestedEventArgs * args) -> HRESULT {
		LPWSTR url;
		UINT32 left, top, width, height;
		args->get_Uri(&url);

		CMFC_webview2Dlg* dlg = new CMFC_webview2Dlg(this);
		dlg->url = url;
		CComPtr<ICoreWebView2Deferral> deferral;
		args->GetDeferral(&deferral);

		dlg->m_onWebViewFirstInitialized = [args, dlg, deferral]() {
			args->put_NewWindow(dlg->m_webView);
			dlg->m_webView->AddWebResourceRequestedFilter(
				L"*", COREWEBVIEW2_WEB_RESOURCE_CONTEXT_ALL);
			dlg->m_webView->add_WebResourceRequested(
				Microsoft::WRL::Callback<ICoreWebView2WebResourceRequestedEventHandler>(
					[](
						ICoreWebView2* sender,
						ICoreWebView2WebResourceRequestedEventArgs* args) {
				COREWEBVIEW2_WEB_RESOURCE_CONTEXT resourceContext;
				args->get_ResourceContext(&resourceContext);
				ICoreWebView2WebResourceRequest* request;
				args->get_Request(&request);

				LPWSTR uri;
				request->get_Uri(&uri);

				LPWSTR method;
				request->get_Method(&method);

				IStream*content;
				request->get_Content(&content);
				return S_OK;
			})
				.Get(), nullptr);
			args->put_Handled(TRUE);
			deferral->Complete();
		};
		if (dlg->Create(IDD_MFC_WEBVIEW2_DIALOG, this))
		{
			dlg->ShowWindow(SW_SHOWNORMAL);
		}
		return S_OK;
	}).Get(), &newWindowRequestedtoken);

	if (m_onWebViewFirstInitialized)
	{
		m_onWebViewFirstInitialized();
		m_onWebViewFirstInitialized = nullptr;
	}

	m_webView->Navigate(url);

	return S_OK;
}
``

**Comment by monica-ch** [comment]
@toolyq Can you also try from our [WebView2 sample](https://github.com/MicrosoftEdge/WebView2Samples/tree/main/SampleApps/WebView2APISample) app?

![image](https://user-images.githubusercontent.com/69160047/218224116-e619f550-3e62-4868-a7d1-097a1970f790.png)

![image](https://user-images.githubusercontent.com/69160047/218227815-2c4b9ff0-f0f1-4f23-9fab-1c1a81e48674.png)

**Comment by toolyq** [comment]
Thanks @monica-ch , 
I think I know the cause:
There has a navigate action in the ICoreWebView2CreateCoreWebView2ControllerCompletedHandler.
![image](https://user-images.githubusercontent.com/12145763/218357216-90798496-9d15-43b7-896d-38642d9cfbaf.png)

But I still don't understand why it used to be work in previous runtime versions.
And if I move the m_webView->Navigate(url); to the position before m_onWebViewFirstInitialized, it also work. 
Maybe the handler orders has changed after some specify version?

**Comment by toolyq** [comment]
@monica-ch , I wandering which version has the behavier changed, do you know about that?
Thanks

**Comment by monica-ch** [comment]
@toolyq We recently fixed an issue where post data is missing during form submission upon (ctrl +click). [See](https://learn.microsoft.com/en-us/microsoft-edge/webview2/release-notes?tabs=dotnetcsharp#bug-fixes)

This is fixed in runtimes 109.0.1496.0+, see #2652.

Are you using ctrl+click action to submit the form? I am seeing the issue (post request is missing) for all the previous version of WebView2 until 108(including 95 and 102) when ctrl+click is used to submit the form. However, this is working fine in 109.0.1496.0+.

**Comment by toolyq** [comment]
We didn't use ctrl+click action, we are using just normal click.

**Comment by monica-ch** [comment]
@toolyq Can I get a small repro app as submitting the form using click is working for me in latest and old versions of WV2.
ctrl+click is working only in the latest version.

**Comment by toolyq** [comment]
@monica-ch,
Here is the repro that can reproduce the normal click issue, it worked in old version WV2, but not latest version unless you update the code of line 224 and 231. I didn't test the ctrl+click problem.
https://github.com/toolyq/webview2-3202/blob/main/MFC_webview2/MFC_webview2/MFC_webview2Dlg.cpp

A sample page here:
https://github.com/toolyq/webview2-3202/blob/main/MFC_webview2/a.html

**Comment by monica-ch** [comment]
@toolyq Thanks for the repro sample. Looks like this was changed in v108.

From the sample, I see you are again navigating to the url for the open windows which is not needed. I recommended to avoid navigation when opening a new window during form submission that causing navigation (form submission with post) to the opened window to abort and navigates again to the url.

Navigating to the opened window url and then invoking the m_onWebViewFirstInitialized in the sample makes the first navigation to ignore and submit the form with post successfully.
